localstack/services/awslambda/event_source_listeners/stream_event_source_listener.py

```diff
@@ -0,0 +1,351 @@
+        end_sequence_num,
+        source_arn,
+        func_arn,
+        invoke_count,
+        status_code,
+        batch_size,
+        first_record_arrival_time,
+        last_record_arrival_time,
+        destination,
+    ):
+        """
+        Creates a metadata payload relating to a failed Lambda invocation and delivers it to the given destination
+        """
+        payload = {
+            "version": "1.0",
+            "timestamp": timestamp_millis(),
+            "requestContext": {
+                "requestId": long_uid(),
+                "functionArn": func_arn,
+                "condition": "RetryAttemptsExhausted",
+                "approximateInvokeCount": invoke_count,
+            },
+            "responseContext": {
+                "statusCode": status_code,
+                "executedVersion": "$LATEST",  # TODO: don't hardcode these fields
+                "functionError": "Unhandled",
+            },
+        }
+        details = {
+            "shardId": shard_id,
+            "startSequenceNumber": start_sequence_num,
+            "endSequenceNumber": end_sequence_num,
+            "approximateArrivalOfFirstRecord": first_record_arrival_time,
+            "approximateArrivalOfLastRecord": last_record_arrival_time,
+            "batchSize": batch_size,
+            "streamArn": source_arn,
+        }
+        payload[self._FAILURE_PAYLOAD_DETAILS_FIELD_NAME] = details
+        send_event_to_target(destination, payload)
+
+    def _monitor_stream_event_sources(self, *args):
+        """
+        Continuously monitors event source mappings. When a new event source for the relevant stream type is created,
+        spawns listener threads for each shard in the stream. When an event source is deleted, stops the associated
+        child threads.
+        """
+        while True:
+            try:
+                # current set of streams + shard IDs that should be feeding Lambda functions based on event sources
+                mapped_shard_ids = set()
+                sources = self._get_matching_event_sources()
+                if not sources:
+                    # Temporarily disable polling if no event sources are configured
+                    # anymore. The loop will get restarted next time a record
+                    # arrives and if an event source is configured.
+                    self._COORDINATOR_THREAD = None
+                    for thread_id in self._STREAM_LISTENER_THREADS:
+                        self._STREAM_LISTENER_THREADS.pop(thread_id)
+                    return
+
+                # make sure each event source stream has a lambda listening on each of its shards
+                for source in sources:
+                    stream_arn = source["EventSourceArn"]
+                    region_name = stream_arn.split(":")[3]
+                    stream_client = self._get_stream_client(region_name)
+                    batch_size = max(min(source.get("BatchSize", 1), 10), 1)
```

Comment:
Q: we currently limit the `batch_size` to 10, as we do not use the `MaximumBatchingWindowInSeconds`, right? 

I found this in the docs: 
> When you set BatchSize to a value greater than 10, you must set MaximumBatchingWindowInSeconds to at least 1.

Reply:
I have no idea why this limit is imposed. I copied this over from the previous implementation. I'm considering removing it because there's no indication for why it exists in the first place, and if we should be imposing a limit silently enforcing it here is probably not the right way to do it.

Reply:
Please at least add a TODO here so we don't forget to fix this :+1: